src/Query/ExecutorFactory.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace React\Dns\Query;
+
+use React\Cache\ArrayCache;
+use React\Cache\CacheInterface;
+use React\Dns\Config\Config;
+use React\Dns\Config\HostsFile;
+use React\EventLoop\Loop;
+use React\EventLoop\LoopInterface;
+
+class ExecutorFactory
```

Comment:
```suggestion
final class ExecutorFactory
```

Reply:
i do not realy understand why a class should in any case a final class? nobody can reuse this class as basic class, this is always the point i got an *facepalm* if i step over this problem...

Reply:
I understand that this may be controversial and agree that *any* decision may not be ideal in any situation, so we're talking tradeoffs here. I don't think we're suggesting *any* class should be `final`, but rather that we would like to have our classes `final` by default unless we have a reason to do otherwise. By being `final`, we ensure that we do not expose any internal assumptions which helps keeping our API surface smaller to avoid any potential BC breaks in the future. (Refs #134)

Reply:
my current use case will be fine with a fine class at this point, but my opinion is to say "no to final" ;) :D